src/main/java/org/opensearch/knn/plugin/KNNPlugin.java

```diff
@@ -338,4 +341,16 @@ public List<NamedXContentRegistry.Entry> getNamedXContent() {
     public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings settings) {
         return ImmutableList.of(new SystemIndexDescriptor(MODEL_INDEX_NAME, "Index for storing models used for k-NN indices"));
     }
+
+    @Override
+    public Settings additionalSettings() {
+        final List<String> engineSettings = Arrays.stream(KNNEngine.values())
```

Comment:
Can you add a comment for what we need to do in this method and why?

Reply:
ack